File: ui/src/main/scala/tripaint/view/JavaFxExt.scala
package tripaint.view

import javafx.scene.paint.Color as FXColor
import tripaint.color.Color

object JavaFxExt {
  extension (c: Color) {
    def toFXColor: FXColor = FXColor.color(clamp(c.r), clamp(c.g), clamp(c.b), clamp(c.a))
  }

  def fromFXColor(c: FXColor): Color = Color(c.getRed, c.getGreen, c.getBlue, c.getOpacity)

  given Conversion[FXColor, Color] with {
    override def apply(c: FXColor): Color = fromFXColor(c)
  }

  private def clamp(v: Double): Double = math.min(math.max(v, 0), 1)
}

File: ui/src/main/kotlin/tripaint/view/JavaFxExt.kt
package tripaint.view

import tripaint.color.Color
import kotlin.math.max
import kotlin.math.min

import javafx.scene.paint.Color as FXColor

object JavaFxExt {
    fun Color.toFXColor(): FXColor = FXColor.color(clamp(this.r), clamp(this.g), clamp(this.b), clamp(this.a))

    fun fromFXColor(c: FXColor): Color = Color(c.red, c.green, c.blue, c.opacity)

    private fun clamp(v: Double): Double = min(max(v, 0.0), 1.0)
}
